r2-core/src/main/java/com/linkedin/r2/transport/http/client/ExponentialBackOffRateLimiter.java

```diff
@@ -173,24 +173,10 @@ public void incrementPeriod()
   @Override
   public void submit(Task t)
   {
-    boolean runNow = false;
     synchronized (this)
     {
-      if (_period == 0 && _pending.isEmpty() && _runningTasks < _maxRunningTasks)
-      {
-        _runningTasks ++;
-        runNow = true;
-      }
-      else
-      {
-        _pending.add(t);
-        schedule();
-      }
-    }
-
-    if (runNow)
-    {
-      t.run(_doneCallback);
+      _pending.add(t);
+      schedule();
```

Comment:
Is this change actually part of your RB?

Reply:
Yes,  to make it more asynchronous and the task submission always happens in the executor thread.